cli/ceph/orch/orch.py

```diff
@@ -0,0 +1,133 @@
+    def apply(self, cluster=None, **kw):
+        """
+        Applies the configuration to hosts.
+        Args:
+          cluster : Ceph cluster
+          kw (Dict): Key/value pairs that needs to be provided to the installer.
+          Example:
+            Supported keys:
+              service-name(str): name of service.
+              hosts(list): name of host(Optional).
+              placement(dict): Placement on hosts(Optional).
+              count(int): No of hosts to be applied(Optional)
+        Returns:
+          Dict(str):  A mapping of host strings of the given task's return
+                      value for that host's execution run.
+        """
+        service_name = kw.pop("service", "")
+        cmd = f"{self.base_cmd} apply {service_name}"
+
+        pos_args = kw.get("pos_args")
+        if pos_args:
+            for arg in pos_args:
+                cmd += f" {arg}"
+
+        args = kw.get("args")
+
+        placement = args.pop("placement", {})
+
+        if placement:
+            placement_str = "--placement="
+            sep = placement.get("sep", " ")
+            node_names = list()
+
+            if "label" in placement:
+                node_names.append(f"label:{placement['label']}")
+
+            if "nodes" in placement:
+                nodes = placement.get("nodes")
+
+                if "*" in nodes:
+                    node_names.append("*")
+                elif "[" in nodes:
+                    placement_str += '"%s"' % nodes
+                else:
+                    nodes_ = get_nodes_by_ids(cluster, nodes)
+                    node_names = [node.hostname for node in nodes_]
+
+            # Support RGW count-per-host placement option
+            if "count-per-host" in placement and service_name == "rgw":
+                node_names.append(f"count-per-host:{placement['count-per-host']}")
+
+            node_str = f"{sep}".join(node_names)
+
+            limit = placement.pop("limit", None)
+            if limit:
+                placement_str += f"'{limit}{sep}{node_str}'"
+            else:
+                placement_str += f"'{node_str}'"
+
+            cmd += placement_str
+
+        # Add the arguments to the cmd
+        if args:
+            cmd += config_dict_to_string(args)
+
+        out, _ = self.execute(sudo=True, cmd=cmd)
+
+        if not out:
+            raise OrchApplyServiceFailure(service_name)
+
+    def rm(self, service_name=None, **kw):
+        """
+        Removes a service.
+        Args:
+          service_name (Str): Name of the service to be removed
+          kw (Dict): Key/value pairs that needs to be provided to the installer.
+          Example:
+            Supported keys:
+              service_name(str): name of service.
+        Returns:
+          Dict(str)
+            A mapping of host strings of the given task's return value for that
+            host's execution run.
+        """
+
+        service = service_name if service_name else kw.pop("service", "")
```

Comment:
same as above if `service_name` is mandatory parameter then it should be part of method signature

Reply:
Adding it here as there are special cases, for eg. the current one where the service name will be iscsi.iscsi whereas the service arg will have only iscsi. So making it as an optional param which gives the flexibility to user to handle such cases

Reply:
Same as above pass mandatory then make this variable as part of method signature